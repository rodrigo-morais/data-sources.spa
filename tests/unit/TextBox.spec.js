import Vuex from 'vuex'
import { mount, createLocalVue } from '@vue/test-utils'
import { createRenderer } from 'vue-server-renderer'

import SearchTextBox from '@/components/SearchArea/Filter/TextBox/index.vue'


const localVue = createLocalVue()
localVue.use(Vuex)

const actions = { fetchDataSources: jest.fn() }
const state = {
  loading: false,
  data: [{ name: 'DataSource 1' }, { name: 'DataSource 2' }],
  error: false,
}
const store = new Vuex.Store({
  state,
  actions,
})

describe('SearchTextBox.vue', () => {
  let wrapper

  beforeEach(() => {
    wrapper = mount(SearchTextBox, { store, localVue })
  })

  it('renders according to design', () => {
    const renderer = createRenderer()
    renderer.renderToString(wrapper.vm, (err, str) => {
      if (err) throw new Error(err)
      expect(str).toMatchSnapshot()
    })
  })

  it('dispatches "fetchDataSources" when input event value has nmore than 3 characters', () => {
    const input = wrapper.find('#searchBox')
    input.element.value = 'data'
    input.trigger('input')

    expect(actions.fetchDataSources).toHaveBeenCalled()
  })
})
