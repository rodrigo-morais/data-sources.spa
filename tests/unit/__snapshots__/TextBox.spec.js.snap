// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchTextBox.vue renders according to design 1`] = `
<form>
  <input type="text" id="searchBox" placeholder="ï€‚ Filter Data Sources...">
</form>
`;
